lib/lf/geometry/segment_o1.cc

```diff
@@ -41,4 +41,66 @@ std::unique_ptr<Geometry> SegmentO1::SubGeometry(dim_t codim, dim_t i) const {
   LF_VERIFY_MSG(false, "codim is out of bounds.");
 }
 
+  std::vector<std::unique_ptr<Geometry>>
+  SegmentO1::ChildGeometry(const RefinementPattern &ref_pat) const {
+    // The refinement pattern must be for a segment
+    LF_VERIFY_MSG(ref_pat.RefEl() == lf::base::RefEl::kSegment(),
+		  "Refinement pattern for " << ref_pat.RefEl().ToString());
+    // Lattice meshwidth
+    const double h_lattice = 1.0/(double)ref_pat.LatticeConst();
+    // Obtain geometry of children as vector of pairs of lattice coordinates
+    std::vector<Eigen::Matrix<int,Eigen::Dynamic,Eigen::Dynamic>>
+      child_polygons(ref_pat.ChildPolygons());
+    // Number of child segments
+    const int no_children = child_polygons.size();
+    std::vector<std::unique_ptr<Geometry>> child_geo_uptrs{};
+    // For each child create a geometry object and a unique pointer to it.
+    for (int l=0; l<no_children; l++) {
+      // A single child must be described by an interval, that is
+      // two differente lattice coordinates 
+      LF_VERIFY_MSG(child_polygons[l].rows() == 1,
```

Comment:
I would suggest to move this check and the next check into the constructor of `RefinementPattern`